server/app/v1/pokemons/mappers/pokemons.mappers.ts

```diff
@@ -0,0 +1,39 @@
+import type {
+  DetailApiResponse,
+  PokemonDetail,
+  PokemonApiResponse,
+  TransformedPokemonList,
+} from '~/type'
+
+export const transformPokemonList = (
+  res: PokemonApiResponse,
```

Comment:
should always only pass the minimum required by the function. the reason is cause when you want to write test for this function, you can just write a fixture to pass to this function on the minimally required data. 

with how you do now, you now need to mock PokemonApiResponse types